### ISSUE TITLE
Enhancing $ref to point to sub elements of a model

### ISSUE THREAD

**Issue by zamrokk** [swagger-api/swagger-ui#2428]
Hi,

I would like swagger ui to be able to point to a sub model element of a known model

like this : 
          $ref: '#/definitions/client/valueProposition/strategicAlignment'

![image](https://cloud.githubusercontent.com/assets/2536368/18908309/6ef84ce2-856f-11e6-9759-9c547cc74486.png)

Is it possible ?

Thanks

**Comment by fehguy** [comment]
No, that's not currently possible but I suppose it could be.  You can currently point to an external document, but not inside the current document like you are showing.  If you could please open an issue in swagger-js we can see about that support.

**Comment by hlubovac** [comment]
Although ugly, I solved this problem for myself by embedding HTML (e.g. links) into swagger-json content, in particular as part of "description" values of corresponding data parts. And, so, I was able to achieve this:

![image](https://cloud.githubusercontent.com/assets/3127736/21822103/2e99d6b4-d733-11e6-995d-3054c17fff11.png)

I think that to fully support linking resources across an API beyond attribute data-types (e.g. for attribute-descriptions or relationship-identifiers) we need some swagger-json specs love. Swagger UI can only present associations within "definitions" via "$ref", but it cannot derive from anywhere that my "case_id", on top of that image above, to me is the identifier of "case" resource that is visually presented elsewhere within swagger-ui. I haven't tried using Markdown syntax within those "description" fields: I only recently discovered that "/info/description" field of swagger-json supports embedding HREF+TEXT data that get rendered as HTML-links. Maybe try that. But, for me, it took a lot of trickery to achieve what's on that image above: server-side swagger-json dynamic generation, customizing HTML, JS and CSS of the swagger-ui. I did get better documentation, but my swagger-ui upgrades are now painful (perhaps, swagger-ui could be enhanced a little to abstract configuration a little better, to minimize customization across all its files).
